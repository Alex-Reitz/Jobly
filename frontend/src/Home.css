.home-heading {
  color: #316685;
  text-align: center;
  font-size: 3.5rem;
  margin-top: 10%;
}
.home-links {
  color: #316685;
  display: flex;
  justify-content: center;
}
.Link {
  text-decoration: none;
  padding: 0.5rem 1rem;
  height: 100%;
  border-bottom: 3px solid transparent;
}
