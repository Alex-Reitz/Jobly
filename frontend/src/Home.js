import React, { useContext } from "react";
import UserContext from "./Context/userContext";
import { Link } from "react-router-dom";

function Home() {
  const user = useContext(UserContext);
  if (!user.username) {
    return (
      <div>
        <h3>Welcome to Jobly!</h3>
        <Link exact to="/signup">
          Signup
        </Link>
        <br />
        <Link exact to="/login">
          Login
        </Link>
      </div>
    );
  } else {
    return (
      <div>
        <h3>Welcome to Jobly {user.username}</h3>
      </div>
    );
  }
}

export default Home;
