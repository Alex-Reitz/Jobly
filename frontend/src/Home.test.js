import React from "react";
import { render } from "@testing-library/react";
import Home from "./Home";
import { MemoryRouter } from "react-router";
import UserContext from "./Context/userContext";

it("matches snapshot", function () {
  const testFunc = jest.fn();
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: testFunc }}>
        <Home />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
