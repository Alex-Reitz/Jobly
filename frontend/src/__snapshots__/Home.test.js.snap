// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="home-content"
  >
    <h3
      class="home-heading"
    >
      Welcome to Jobly!
    </h3>
    <div
      class="home-links"
    >
      <a
        class="Link"
        href="/signup"
      >
        Signup
      </a>
      <br />
      <a
        class="Link"
        href="/login"
      >
        Login
      </a>
    </div>
  </div>
</DocumentFragment>
`;
